tests/shutdown/cleanup_before_shutdown.pm

```diff
@@ -42,7 +43,7 @@ END_SCRIPT
         assert_script_run $script;
         assert_script_run "chmod +x /usr/lib/systemd/system-shutdown/debug.sh";
     }
-    if (get_var('DROP_PERSISTENT_NET_RULES')) {
+    if (get_var('DROP_PERSISTENT_NET_RULES') && is_qemu) {
```

Comment:
additionally to this we could also check for the right place for `DROP_PERSISTENT_NET_RULES` in openQA . Looks like that currently it is define in too broad place so we should move to place where it will apply **only** for QEMU jobs

Reply:
Deleting `70-persistent-net.rules` should be possible only on qemu **or** it has to be done on qemu ?

Reply:
I guess that it's more about the jobs where a hard drive is being exported for later reuse (qemu, hyperv and svirt jobs), which does make a lot of sense. And the drop could be by default... only keeping it, when it's actually wanted...

Reply:
IMHO having variable and then allowing it only in QEMU will be confusing. Although are you sure somebody will not need it for non-qemu for some reason? THus If problematic on other backends, could be solved with proper setup?

Reply:
> additionally to this we could also check for the right place for `DROP_PERSISTENT_NET_RULES` in openQA . Looks like that currently it is define in too broad place so we should move to place where it will apply **only** for QEMU jobs

Do you have an idea where `QEMU only` place could be? Same thing could be applied as well to wicked cleanup, which is few lines bellow. It could be moved there, but maybe better to have it separated.

Reply:
> I guess that it's more about the jobs where a hard drive is being exported for later reuse (qemu, hyperv and svirt jobs), which does make a lot of sense. And the drop could be by default... only keeping it, when it's actually wanted...

You mean to reverse the logic?  Delete rules by default and have variable `KEEP_PERSISTENT_NET_RULES`?

Reply:
> IMHO having variable and then allowing it only in QEMU will be confusing. Although are you sure somebody will not need it for non-qemu for some reason? THus If problematic on other backends, could be solved with proper setup?

Well the original problem comes from bad settings, where variable is/was set at medium level. This will be moved to test suite setting, but i still think that original intention was focused on image.

Reply:
Well, I'm not against to have it on targeted on medium level. IMHO it'd be better to rename the variable to address changed functionality. But understand that changing variable name cost much more effort.

Reply:
> > I guess that it's more about the jobs where a hard drive is being exported for later reuse (qemu, hyperv and svirt jobs), which does make a lot of sense. And the drop could be by default... only keeping it, when it's actually wanted...
> 
> You mean to reverse the logic? Delete rules by default and have variable `KEEP_PERSISTENT_NET_RULES`?

Yep. Afterall, `(!KEEP_PERSISTENT_NET_RULES && (is_qemu && is_hyperv && is_svirt)` or something like this

Reply:
> Yep. Afterall, `(!KEEP_PERSISTENT_NET_RULES && (is_qemu && is_hyperv && is_svirt)` or something like this

Thx, I will think about it, so at the end we will will drop the drop.

Reply:
Yeah, ~~tbh... the net rules should be autogenerated if they're not present... the "extra" safety of only doing it under certain backends its overkill for me~~

Reply:
Now I see that `hyperv` is just `BACKEND=svirt`, there is no need for hyperv check.